### PR TITLE
Current period transaction closure basic validation

### PR DESCRIPTION
**Pull request by amoriartyCH**
-Added a new directory too house all future transaction closure validators
-Added basic current period transaction closure validation

-Added test classes.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/AccountsValidator.java [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+import uk.gov.companieshouse.api.accounts.validation.transactionclosure.CurrentPeriodTnClosureValidator;
 import uk.gov.companieshouse.api.model.transaction.Transaction;
 
 import javax.servlet.http.HttpServletRequest;
 
 @Component
 public class AccountsValidator {
 
+    @Autowired
+    CurrentPeriodTnClosureValidator currentPeriodTnClosureValidator;
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/validation/AccountsValidator.java:16]: Could this be private? (Sorry)

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/CurrentPeriodTnClosureValidator.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        Map<String, String> companyAccountLinks;
+        Map<String, String> smallFullLinks;
+
+        ResponseObject<CompanyAccount> companyAccountResponseObject = companyAccountService.findById(companyAccountsId, request);
+        companyAccountLinks = companyAccountResponseObject.getData().getLinks();
+
+        if (companyAccountLinks.get(CompanyAccountLinkType.SMALL_FULL.getLink()) != null) {
+            smallFullLinks = smallFullService.find(companyAccountsId, request).getData().getLinks();
+
+            if (!smallFullLinks.isEmpty() && smallFullLinks.get(SmallFullLinkType.CURRENT_PERIOD.getLink()) != null) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/CurrentPeriodTnClosureValidator.java:61]: If the current_period link is absent, we ought to return the mandatory_element_missing error; no need to fetch the current period resource

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/AccountsValidator.java (hunk 1 above)
Comment (amoriartyCH) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/AccountsValidator.java]: Added, i almost got away with it hahaha

File: src/main/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/CurrentPeriodTnClosureValidator.java (hunk 2 above)
Comment (amoriartyCH) [reply src/main/java/uk/gov/companieshouse/api/accounts/validation/transactionclosure/CurrentPeriodTnClosureValidator.java]: changed logic, new logic doesn't pull the resource back, just checks the if the link is missing, if it is it adds the error.

**Review by BSaunders95 — APPROVED** [review]

**Review by BSaunders95 — APPROVED** [review]

**Pull request merged**